client/src/components/LinkButton.jsx

```diff
@@ -1,48 +1,84 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import propTypes from 'prop-types';
 import Button from 'plaid-threads/Button';
 import Touchable from 'plaid-threads/Touchable';
 import { usePlaidLink } from 'react-plaid-link';
 
 LinkButton.propTypes = {
   config: propTypes.object,
-  update: propTypes.bool,
+  isUpdate: propTypes.bool,
+  isOauth: propTypes.bool,
+  userId: propTypes.number,
+  itemId: propTypes.number || null,
 };
 
 LinkButton.defaultProps = {
   config: null,
-  update: false,
+  isUpdate: false,
+  isOauth: false,
+  userId: null,
+  itemId: null,
 };
 
-export default function LinkButton({ children, config, update }) {
+export default function LinkButton({
+  isOauth,
+  children,
+  config,
+  isUpdate,
+  userId,
+  itemId,
+}) {
+  // add additional receivedRedirectUri config when re-launching Link in Oauth.
+  if (isOauth) {
+    config.receivedRedirectUri = window.location.href;
+  }
   const { open, ready } = usePlaidLink(config);
 
+  useEffect(() => {
+    // initiallizes Link automatically if it is OAuth
```

Comment:
(nit) this is a little ambiguous as "is OAuth" could mean either before or after redirect

```suggestion
    // initiallizes Link automatically if it is handling an OAuth redirect
```